### PR TITLE
Allow multiple values_files

### PR DESCRIPTION
**Pull request by betoredhat**
Some charts requires mutiple custom files with configuration values

### CODE REVIEW

File: roles/chart_verifier/tasks/tests.yml [hunk 1]
```diff
@@ -90,29 +90,25 @@
         - chart_stat is defined
         - chart_stat.stat.exists
 
-    - name: "Check values_file stat"
+    - name: "Check if chart.values_file exist"
       stat:
```
Comment (nsilla) [inline roles/chart_verifier/tasks/tests.yml:94]: Hi Beto,
What if the values_file list include a web URL? This task will return false even if the URL is valid.

File: roles/chart_verifier/tasks/tests.yml [hunk 2]
```diff
@@ -90,29 +90,25 @@
...
-        path: "{{ chart.values_file }}"
-      register: values_stat
-      when:
```
Comment (nsilla) [inline roles/chart_verifier/tasks/tests.yml:97]: Also, I guess we should keep the "when" condition here as long as the values_file variable is optional.

File: roles/chart_verifier/tasks/tests.yml [hunk 3]
```diff
@@ -140,7 +136,11 @@
       -v {{ work_chart_dir }}:/app/chartverifier:Z
       {{ chart_verifier_image }}
       verify --set profile.vendorType=partner
-      {% if chart.values_file is defined %} -F /app/chartverifier/{{ chart.values_file | basename }} {% endif %}
+      {%- if chart.values_file is defined %}
```
Comment (ramperher) [inline roles/chart_verifier/tasks/tests.yml:139]: Should we check here that chart.values_file is iterable?

**Review by ramperher — COMMENTED** [review]
One comment from my side

File: roles/chart_verifier/tasks/tests.yml (hunk 2 above)
Comment (betoredhat) [reply roles/chart_verifier/tasks/tests.yml:97]: This block is only is applied only for local files. See line 113.

File: roles/chart_verifier/tasks/tests.yml (hunk 3 above)
Comment (betoredhat) [reply roles/chart_verifier/tasks/tests.yml:153]: Thanks, good idea. Added the validation at the beginning

File: roles/chart_verifier/tasks/tests.yml (hunk 1 above)
Comment (betoredhat) [reply roles/chart_verifier/tasks/tests.yml:104]: This block is to check local files

File: roles/chart_verifier/tasks/tests.yml [hunk 4]
```diff
@@ -34,35 +43,36 @@
       when:
         - file_response.status | int != 200
 
-    - name: "Check if the chart values file exists"
+    - name: "Download chart file"
+      get_url:
+        url: "{{ chart.chart_file }}"
+        dest: "{{ work_chart_dir }}/"
+        mode: "0750"
```
Comment (tonyskapunk) [inline roles/chart_verifier/tasks/tests.yml:50]: Don't think we need it +w, can we do 0640?

File: roles/chart_verifier/tasks/tests.yml [hunk 5]
```diff
@@ -90,29 +100,29 @@
...
         - chart.values_file is defined
-        - values_stat.stat.exists | bool == false
+        - item.stat.exists == false
 
     - name: "Copy values_file to working directory"
       copy:
-        src: "{{ chart.values_file }}"
+        src: "{{ item }}"
         dest: "{{ work_chart_dir }}/"
+      loop: "{{ chart.values_file }}"
```
Comment (tonyskapunk) [inline roles/chart_verifier/tasks/tests.yml:123]: Is this one looping through a list that could have URLs? If so, would need to take the basename of the URL

File: roles/chart_verifier/tasks/tests.yml (hunk 4 above)
Comment (betoredhat) [reply roles/chart_verifier/tasks/tests.yml:50]: done

File: roles/chart_verifier/tasks/tests.yml (hunk 5 above)
Comment (betoredhat) [reply roles/chart_verifier/tasks/tests.yml:123]: This block is to process local files. The chart_file location determines if the files are local or remote. If the chart_file is local, the values files must be local too. If the chart file is remote, the files must be remote. That has been clarified in the readme and is the use case exposed by the requester.

Maybe in the next iteration, we can add support for files in remote (uri) and local (copy/stat).

Thanks

**Review by tonyskapunk — APPROVED** [review]
LGTM, thanks!

**Pull request merged**